rust-demangle.c

```diff
@@ -791,11 +939,47 @@ static void demangle_const_uint(struct rust_demangler *rdm) {
     // Print anything that doesn't fit in `uint64_t` verbatim.
     if (hex_len > 16) {
         PRINT("0x");
-        print_str(rdm, rdm->sym + (rdm->next - hex_len), hex_len);
-        return;
+        print_str(rdm, rdm->sym + (rdm->next - hex_len - 1), hex_len);
+    } else {
+        print_uint64(rdm, value);
+    }
+
+    if (rdm->verbose)
+        PRINT(basic_type(ty_tag));
+}
+
+
+static void demangle_const_str_literal(struct rust_demangler *rdm) {
+    CHECK_OR(!rdm->errored, return );
+
+    PRINT("\"");
+
+    // FIXME(bjorn3) actually decode UTF-8 strings into individual characters
```

Comment:
What's kind of funny is that 1. `try_parse_str_chars` in the Rust version barely and 2. there's a UTF-8 *encoder* somewhere in this file, for punycode, so I feel like combining the two (the encoder giving you the mapping of the bits to bytes) might not even be that hard. And we have test data, so I can golf it if you want.